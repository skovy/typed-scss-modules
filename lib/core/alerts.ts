import chalk from "chalk";

export const LOG_LEVELS = ["verbose", "error", "info", "silent"] as const;
export type LogLevel = typeof LOG_LEVELS[number];

export const logLevelDefault: LogLevel = "verbose";

let currentLogLevel: LogLevel | undefined;

export const setAlertsLogLevel = (logLevel: LogLevel) => {
  currentLogLevel = logLevel;
};

type CbFunc = (...args: any[]) => any;
type WrappedCbFunc<T extends CbFunc> = (
  ...args: Parameters<T>
) => ReturnType<T> | void;
/**
 * wraps a callback and only calls it if currentLogLevel is undefined or included in permittedLogLevels
 * @param permittedLogLevels list of log levels. callbacks will only be called if current log level is listed here
 * @param cb callback
 */
const withLogLevelsRestriction =
  <T extends CbFunc>(permittedLogLevels: LogLevel[], cb: T): WrappedCbFunc<T> =>
  (...args: Parameters<T>): ReturnType<T> | void => {
    const shouldCall =
      !currentLogLevel || permittedLogLevels.includes(currentLogLevel);

    if (shouldCall) {
      return cb(...args);
    }
  };

const error = withLogLevelsRestriction(
  ["verbose", "error", "info"],
  (message: string) => console.log(chalk.red(message))
);
const warn = withLogLevelsRestriction(["verbose"], (message: string) =>
  console.log(chalk.yellowBright(message))
);
const notice = withLogLevelsRestriction(
  ["verbose", "info"],
  (message: string) => console.log(chalk.gray(message))
);
const info = withLogLevelsRestriction(["verbose", "info"], (message: string) =>
  console.log(chalk.blueBright(message))
);
const success = withLogLevelsRestriction(
  ["verbose", "info"],
  (message: string) => console.log(chalk.green(message))
);

export const alerts = { error, warn, notice, info, success };
