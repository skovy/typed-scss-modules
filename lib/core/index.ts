export { alerts, setAlertsLogLevel } from "./alerts";
export { generate } from "./generate";
export { listDifferent } from "./list-different";
export { CLIOptions, ConfigOptions } from "./types";
export { watch } from "./watch";
export { writeFile } from "./write-file";
