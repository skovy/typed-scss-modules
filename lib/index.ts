export { main as default } from "./main";
