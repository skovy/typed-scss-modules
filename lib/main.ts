import fs from "fs";
import path from "path";
import slash from "slash";

import {
  watch,
  MainOptions,
  generate,
  listDifferent,
  setAlertsLogLevel,
} from "./core";

export const main = async (pattern: string, options: MainOptions) => {
  setAlertsLogLevel(options.logLevel);

  // When the provided pattern is a directory construct the proper glob to find
  // all .scss files within that directory. Also, add the directory to the
  // included paths so any imported with a path relative to the root of the
  // project still works as expected without adding many include paths.
  if (fs.existsSync(pattern) && fs.lstatSync(pattern).isDirectory()) {
    if (Array.isArray(options.includePaths)) {
      options.includePaths.push(pattern);
    } else {
      options.includePaths = [pattern];
    }

    // When the pattern provide is a directory, assume all .scss files within.
    pattern = slash(path.resolve(pattern, "**/*.scss"));
  }

  if (options.listDifferent) {
    listDifferent(pattern, options);
    return;
  }

  if (options.watch) {
    watch(pattern, options);
  } else {
    await generate(pattern, options);
  }
};
