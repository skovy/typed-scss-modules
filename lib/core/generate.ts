import glob from "glob";

import { alerts } from "./alerts";
import { ConfigOptions } from "./types";
import { writeFile } from "./write-file";

/**
 * Given a file glob generate the corresponding types once.
 *
 * @param pattern the file pattern to generate type definitions for
 * @param options the CLI options
 */
export const generate = async (
  pattern: string,
  options: ConfigOptions
): Promise<void> => {
  // Find all the files that match the provided pattern.
  const files = glob.sync(pattern, { ignore: options.ignore });

  if (!files || !files.length) {
    alerts.error("No files found.");
    return;
  }

  // This case still works as expected but it's easy to do on accident so
  // provide a (hopefully) helpful warning.
  if (files.length === 1) {
    alerts.warn(
      `Only 1 file found for ${pattern}. If using a glob pattern (eg: dir/**/*.scss) make sure to wrap in quotes (eg: "dir/**/*.scss").`
    );
  }

  alerts.success(
    `Found ${files.length} file${
      files.length === 1 ? `` : `s`
    }. Generating type definitions...`
  );

  // Wait for all the type definitions to be written.
  await Promise.all(files.map((file) => writeFile(file, options)));
};
