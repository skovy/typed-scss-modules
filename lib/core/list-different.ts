import glob from "glob";
import fs from "fs";

import { alerts } from "./alerts";
import { CLIOptions } from "./types";
import { fileToClassNames } from "../sass";
import {
  classNamesToTypeDefinitions,
  getTypeDefinitionPath,
} from "../typescript";

export const listDifferent = async (
  pattern: string,
  options: CLIOptions
): Promise<void> => {
  // Find all the files that match the provided pattern.
  const files = glob.sync(pattern);

  if (!files || !files.length) {
    alerts.notice("No files found.");
    return;
  }

  // Wait for all the files to be checked.
  await Promise.all(files.map((file) => checkFile(file, options))).then(
    (results) => {
      results.includes(false) && process.exit(1);
    }
  );
};

export const checkFile = (
  file: string,
  options: CLIOptions
): Promise<boolean> => {
  return new Promise((resolve) =>
    fileToClassNames(file, options)
      .then(async (classNames) => {
        const typeDefinition = await classNamesToTypeDefinitions({
          classNames: classNames,
          ...options,
        });

        if (!typeDefinition) {
          // Assume if no type defs are necessary it's fine
          resolve(true);
          return;
        }

        const path = getTypeDefinitionPath(file);

        if (!fs.existsSync(path)) {
          alerts.error(
            `[INVALID TYPES] Type file needs to be generated for ${file} `
          );
          resolve(false);
          return;
        }

        const content = fs.readFileSync(path, { encoding: "utf8" });

        if (content !== typeDefinition) {
          alerts.error(`[INVALID TYPES] Check type definitions for ${file}`);
          resolve(false);
          return;
        }

        resolve(true);
      })
      .catch((error) => {
        alerts.error(
          `An error occurred checking ${file}:\n${JSON.stringify(error)}`
        );
        resolve(false);
      })
  );
};
