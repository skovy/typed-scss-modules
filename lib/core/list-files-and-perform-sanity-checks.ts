import glob from "glob";

import { alerts } from "./alerts";
import { ConfigOptions } from "./types";

/**
 * Return the files matching the given pattern and alert the user if only 0 or 1
 * files matched.
 *
 * @param pattern the file pattern to generate type definitions for
 * @param options the CLI options
 */
export function listFilesAndPerformSanityChecks(
  pattern: string,
  options: ConfigOptions
): string[] {
  // Find all the files that match the provided pattern.
  const files = glob.sync(pattern, { ignore: options.ignore });

  if (!files || !files.length) {
    alerts.error("No files found.");
  }

  // This case still works as expected but it's easy to do on accident so
  // provide a (hopefully) helpful warning.
  if (files.length === 1) {
    alerts.warn(
      `Only 1 file found for ${pattern}. If using a glob pattern (eg: dir/**/*.scss) make sure to wrap in quotes (eg: "dir/**/*.scss").`
    );
  }

  return files;
}
