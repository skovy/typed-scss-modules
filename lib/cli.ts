#!/usr/bin/env node

import yargs from "yargs";

import { Aliases, NAME_FORMATS } from "./sass";
import { EXPORT_TYPES, QUOTE_TYPES, LOG_LEVELS } from "./typescript";
import { main } from "./main";
import { IMPLEMENTATIONS } from "./implementations";

const { _: patterns, ...rest } = yargs
  .usage(
    "Generate .scss.d.ts from CSS module .scss files.\nUsage: $0 <glob pattern> [options]"
  )
  .example("$0 src", "All .scss files at any level in the src directory")
  .example(
    "$0 src/**/*.scss",
    "All .scss files at any level in the src directory"
  )
  .example(
    "$0 src/**/*.scss --watch",
    "Watch all .scss files at any level in the src directory that are added or changed"
  )
  .example(
    "$0 src/**/*.scss --includePaths src/core src/variables",
    'Search the "core" and "variables" directory when resolving imports'
  )
  .example(
    "$0 src/**/*.scss --aliases.~name variables",
    'Replace all imports for "~name" with "variables"'
  )
  .example(
    "$0 src/**/*.scss --aliasPrefixes.~ ./node_modules/",
    'Replace the "~" prefix with "./node_modules/" for all imports beginning with "~"'
  )
  .example(
    "$0 src/**/*.scss --ignore **/secret.scss",
    'Ignore any file names "secret.scss"'
  )
  .example(
    "$0 src/**/*.scss --implementation sass",
    "Use the Dart SASS package"
  )
  .example(
    "$0 src/**/*.scss -e default --quoteType double",
    "Use double quotes around class name definitions rather than single quotes."
  )
  .example("$0 src/**/*.scss --logLevel error", "Output only errors")
  .demandCommand(1)
  .option("additionalData", {
    string: true,
    alias: "d",
    describe: "Prepends the SCSS code before each file.",
  })
  .option("aliases", {
    coerce: (obj): Aliases => obj,
    alias: "a",
    describe: "Alias any import to any other value.",
  })
  .option("aliasPrefixes", {
    coerce: (obj): Aliases => obj,
    alias: "p",
    describe: "A prefix for any import to rewrite to another value.",
  })
  .option("nameFormat", {
    choices: NAME_FORMATS,
    alias: "n",
    describe: "The name format that should be used to transform class names.",
  })
  .option("implementation", {
    choices: IMPLEMENTATIONS,
    describe:
      "The SASS package to used to compile. This will default to the sass implementation you have installed.",
  })
  .option("exportType", {
    choices: EXPORT_TYPES,
    alias: "e",
    describe: "The type of export used for defining the type definitions.",
  })
  .option("exportTypeName", {
    string: true,
    describe:
      'Set a custom type name for styles when --exportType is "default."',
  })
  .option("exportTypeInterface", {
    string: true,
    describe:
      'Set a custom interface name for styles when --exportType is "default."',
  })
  .option("watch", {
    boolean: true,
    alias: "w",
    describe:
      "Watch for added or changed files and (re-)generate the type definitions.",
  })
  .option("ignoreInitial", {
    boolean: true,
    describe: "Skips the initial build when passing the watch flag.",
  })
  .option("listDifferent", {
    boolean: true,
    alias: "l",
    describe:
      "List any type definitions that are different than those that would be generated.",
  })
  .option("includePaths", {
    array: true,
    string: true,
    alias: "i",
    describe: "Additional paths to include when trying to resolve imports.",
  })
  .option("ignore", {
    string: true,
    array: true,
    describe: "Add a pattern or an array of glob patterns to exclude matches.",
  })
  .option("outputFolder", {
    string: true,
    alias: "o",
    describe:
      "Define a (relative) folder to output the generated type definitions. Note this requires adding the output folder to tsconfig.json `rootDirs`.",
  })
  .options("quoteType", {
    choices: QUOTE_TYPES,
    alias: "q",
    describe:
      "Specify the quote type so that generated files adhere to your TypeScript rules.",
  })
  .options("updateStaleOnly", {
    boolean: true,
    alias: "u",
    describe:
      "Overwrite generated files only if the source file has more recent changes.",
  })
  .option("logLevel", {
    choices: LOG_LEVELS,
    alias: "L",
    describe: "Verbosity level of console output",
  })
  .options("banner", {
    string: true,
    describe:
      "Inserts text at the top of every output file for documentation purposes.",
  })
  .parseSync();

main(patterns[0] as string, { ...rest });
