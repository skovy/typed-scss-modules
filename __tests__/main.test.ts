import fs from "fs";
import path from "path";
import slash from "slash";

import { alerts } from "../lib/core";
import { main } from "../lib/main";
import { describeAllImplementations } from "./helpers";

describeAllImplementations((implementation) => {
  describe("main", () => {
    let writeFileSyncSpy: jest.SpyInstance;

    beforeEach(() => {
      // Only mock the writes, so the example files can still be read.
      writeFileSyncSpy = jest.spyOn(fs, "writeFileSync").mockImplementation();

      // Avoid creating directories while running tests.
      jest.spyOn(fs, "mkdirSync").mockImplementation();

      // Avoid console logs showing up.
      jest.spyOn(console, "log").mockImplementation();

      jest.spyOn(alerts, "error").mockImplementation();
    });

    afterEach(() => {
      writeFileSyncSpy.mockReset();
    });

    test("generates types for all .scss files when the pattern is a directory", async () => {
      const pattern = `${__dirname}/dummy-styles`;

      await main(pattern, {
        banner: "",
        watch: false,
        ignoreInitial: false,
        exportType: "named",
        exportTypeName: "ClassNames",
        exportTypeInterface: "Styles",
        listDifferent: false,
        ignore: [],
        implementation,
        quoteType: "single",
        updateStaleOnly: false,
        logLevel: "verbose",
        additionalData: "$global-red: red;",
        aliases: {
          "~fancy-import": "complex",
          "~another": "style",
        },
        aliasPrefixes: {
          "~": "nested-styles/",
        },
      });

      expect(alerts.error).not.toHaveBeenCalled();
      expect(fs.writeFileSync).toBeCalledTimes(9);

      const expectedDirname = slash(path.join(__dirname, "dummy-styles"));

      expect(fs.writeFileSync).toBeCalledWith(
        `${expectedDirname}/complex.scss.d.ts`,
        "export declare const nestedAnother: string;\nexport declare const nestedClass: string;\nexport declare const number1: string;\nexport declare const someStyles: string;\nexport declare const whereSelector: string;\n"
      );
      expect(fs.writeFileSync).toBeCalledWith(
        `${expectedDirname}/style.scss.d.ts`,
        "export declare const someClass: string;\n"
      );
    });

    test("generates types for all .scss files and ignores files that match the ignore pattern", async () => {
      const pattern = `${__dirname}/dummy-styles`;

      await main(pattern, {
        banner: "",
        watch: false,
        ignoreInitial: false,
        exportType: "named",
        exportTypeName: "ClassNames",
        exportTypeInterface: "Styles",
        listDifferent: false,
        ignore: ["**/style.scss"],
        implementation,
        quoteType: "single",
        updateStaleOnly: false,
        logLevel: "verbose",
        additionalData: "$global-red: red;",
        aliases: {
          "~fancy-import": "complex",
          "~another": "style",
        },
        aliasPrefixes: {
          "~": "nested-styles/",
        },
      });

      expect(alerts.error).not.toHaveBeenCalled();
      expect(fs.writeFileSync).toBeCalledTimes(7);

      const expectedDirname = slash(path.join(__dirname, "dummy-styles"));

      expect(fs.writeFileSync).toBeCalledWith(
        `${expectedDirname}/complex.scss.d.ts`,
        "export declare const nestedAnother: string;\nexport declare const nestedClass: string;\nexport declare const number1: string;\nexport declare const someStyles: string;\nexport declare const whereSelector: string;\n"
      );

      // Files that should match the ignore pattern.
      expect(fs.writeFileSync).not.toBeCalledWith(
        `${expectedDirname}/style.scss.d.ts`,
        expect.anything()
      );
      expect(fs.writeFileSync).not.toBeCalledWith(
        `${expectedDirname}/nested-styles/style.scss.d.ts`,
        expect.anything()
      );
    });

    test("reads options from the configuration file", async () => {
      const pattern = `${__dirname}/dummy-styles`;

      jest.spyOn(process, "cwd").mockReturnValue(path.resolve(pattern));

      await main(pattern, {
        additionalData: "$global-red: red;",
        aliases: {
          "~fancy-import": "complex",
          "~another": "style",
        },
        aliasPrefixes: {
          "~": "nested-styles/",
        },
        exportType: "default",
      });

      expect(alerts.error).not.toHaveBeenCalled();
      expect(fs.writeFileSync).toBeCalledTimes(9);
      // Transform the calls into a more readable format for the snapshot.
      const contents = writeFileSyncSpy.mock.calls
        .map(([fullFilePath, contents]) => ({
          path: path.relative(__dirname, fullFilePath),
          contents,
        }))
        // Sort to avoid flakey snapshot tests if call order changes.
        .sort((a, b) => a.path.localeCompare(b.path));
      expect(contents).toMatchSnapshot();
    });

    test("outputs the correct files when outputFolder is passed", async () => {
      const pattern = path.resolve(__dirname, "dummy-styles");

      await main(pattern, {
        additionalData: "$global-red: red;",
        aliases: {
          "~fancy-import": "complex",
          "~another": "style",
        },
        aliasPrefixes: {
          "~": "nested-styles/",
        },
        outputFolder: "__generated__",
      });

      expect(alerts.error).not.toHaveBeenCalled();
      expect(fs.writeFileSync).toBeCalledTimes(9);
      expect(fs.mkdirSync).toBeCalledTimes(9);
      // Transform the calls into a more readable format for the snapshot.
      const contents = writeFileSyncSpy.mock.calls
        .map(([fullFilePath, contents]) => ({
          path: path.relative(__dirname, fullFilePath),
          contents,
        }))
        // Sort to avoid flakey snapshot tests if call order changes.
        .sort((a, b) => a.path.localeCompare(b.path));
      expect(contents).toMatchSnapshot();
    });
  });
});
