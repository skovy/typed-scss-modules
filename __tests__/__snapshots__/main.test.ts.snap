// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`node-sass implementation main outputs the correct files when outputFolder is passed 1`] = `
Array [
  Object {
    "contents": "export declare const myCustomClass: string;
export declare const nestedAnother: string;
export declare const nestedClass: string;
export declare const nestedStyles: string;
export declare const number1: string;
export declare const someStyles: string;
export declare const whereSelector: string;
",
    "path": "../__generated__/__tests__/dummy-styles/alias-prefixes.scss.d.ts",
  },
  Object {
    "contents": "export declare const myCustomClass: string;
export declare const nestedAnother: string;
export declare const nestedClass: string;
export declare const number1: string;
export declare const someClass: string;
export declare const someStyles: string;
export declare const whereSelector: string;
",
    "path": "../__generated__/__tests__/dummy-styles/aliases.scss.d.ts",
  },
  Object {
    "contents": "export declare const nestedAnother: string;
export declare const nestedClass: string;
export declare const number1: string;
export declare const someStyles: string;
export declare const whereSelector: string;
",
    "path": "../__generated__/__tests__/dummy-styles/complex.scss.d.ts",
  },
  Object {
    "contents": "export declare const composedClass: string;
",
    "path": "../__generated__/__tests__/dummy-styles/composes.scss.d.ts",
  },
  Object {
    "contents": "export declare const app: string;
export declare const appHeader: string;
export declare const logo: string;
",
    "path": "../__generated__/__tests__/dummy-styles/dashes.scss.d.ts",
  },
  Object {
    "contents": "export declare const globalStyle: string;
",
    "path": "../__generated__/__tests__/dummy-styles/global-variables.scss.d.ts",
  },
  Object {
    "contents": "export declare const randomClass: string;
",
    "path": "../__generated__/__tests__/dummy-styles/invalid.scss.d.ts",
  },
  Object {
    "contents": "export declare const nestedStyles: string;
",
    "path": "../__generated__/__tests__/dummy-styles/nested-styles/style.scss.d.ts",
  },
  Object {
    "contents": "export declare const someClass: string;
",
    "path": "../__generated__/__tests__/dummy-styles/style.scss.d.ts",
  },
]
`;

exports[`node-sass implementation main reads options from the configuration file 1`] = `
Array [
  Object {
    "contents": "export type Styles = {
  myCustomClass: string;
  nestedAnother: string;
  nestedClass: string;
  nestedStyles: string;
  number1: string;
  someStyles: string;
  whereSelector: string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
",
    "path": "dummy-styles/alias-prefixes.scss.d.ts",
  },
  Object {
    "contents": "export type Styles = {
  myCustomClass: string;
  nestedAnother: string;
  nestedClass: string;
  number1: string;
  someClass: string;
  someStyles: string;
  whereSelector: string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
",
    "path": "dummy-styles/aliases.scss.d.ts",
  },
  Object {
    "contents": "export type Styles = {
  nestedAnother: string;
  nestedClass: string;
  number1: string;
  someStyles: string;
  whereSelector: string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
",
    "path": "dummy-styles/complex.scss.d.ts",
  },
  Object {
    "contents": "export type Styles = {
  composedClass: string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
",
    "path": "dummy-styles/composes.scss.d.ts",
  },
  Object {
    "contents": "export type Styles = {
  app: string;
  appHeader: string;
  logo: string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
",
    "path": "dummy-styles/dashes.scss.d.ts",
  },
  Object {
    "contents": "export type Styles = {
  globalStyle: string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
",
    "path": "dummy-styles/global-variables.scss.d.ts",
  },
  Object {
    "contents": "export type Styles = {
  randomClass: string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
",
    "path": "dummy-styles/invalid.scss.d.ts",
  },
  Object {
    "contents": "export type Styles = {
  nestedStyles: string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
",
    "path": "dummy-styles/nested-styles/style.scss.d.ts",
  },
  Object {
    "contents": "export type Styles = {
  someClass: string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
",
    "path": "dummy-styles/style.scss.d.ts",
  },
]
`;

exports[`sass implementation main outputs the correct files when outputFolder is passed 1`] = `
Array [
  Object {
    "contents": "export declare const myCustomClass: string;
export declare const nestedAnother: string;
export declare const nestedClass: string;
export declare const nestedStyles: string;
export declare const number1: string;
export declare const someStyles: string;
export declare const whereSelector: string;
",
    "path": "../__generated__/__tests__/dummy-styles/alias-prefixes.scss.d.ts",
  },
  Object {
    "contents": "export declare const myCustomClass: string;
export declare const nestedAnother: string;
export declare const nestedClass: string;
export declare const number1: string;
export declare const someClass: string;
export declare const someStyles: string;
export declare const whereSelector: string;
",
    "path": "../__generated__/__tests__/dummy-styles/aliases.scss.d.ts",
  },
  Object {
    "contents": "export declare const nestedAnother: string;
export declare const nestedClass: string;
export declare const number1: string;
export declare const someStyles: string;
export declare const whereSelector: string;
",
    "path": "../__generated__/__tests__/dummy-styles/complex.scss.d.ts",
  },
  Object {
    "contents": "export declare const composedClass: string;
",
    "path": "../__generated__/__tests__/dummy-styles/composes.scss.d.ts",
  },
  Object {
    "contents": "export declare const app: string;
export declare const appHeader: string;
export declare const logo: string;
",
    "path": "../__generated__/__tests__/dummy-styles/dashes.scss.d.ts",
  },
  Object {
    "contents": "export declare const globalStyle: string;
",
    "path": "../__generated__/__tests__/dummy-styles/global-variables.scss.d.ts",
  },
  Object {
    "contents": "export declare const randomClass: string;
",
    "path": "../__generated__/__tests__/dummy-styles/invalid.scss.d.ts",
  },
  Object {
    "contents": "export declare const nestedStyles: string;
",
    "path": "../__generated__/__tests__/dummy-styles/nested-styles/style.scss.d.ts",
  },
  Object {
    "contents": "export declare const someClass: string;
",
    "path": "../__generated__/__tests__/dummy-styles/style.scss.d.ts",
  },
]
`;

exports[`sass implementation main reads options from the configuration file 1`] = `
Array [
  Object {
    "contents": "export type Styles = {
  myCustomClass: string;
  nestedAnother: string;
  nestedClass: string;
  nestedStyles: string;
  number1: string;
  someStyles: string;
  whereSelector: string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
",
    "path": "dummy-styles/alias-prefixes.scss.d.ts",
  },
  Object {
    "contents": "export type Styles = {
  myCustomClass: string;
  nestedAnother: string;
  nestedClass: string;
  number1: string;
  someClass: string;
  someStyles: string;
  whereSelector: string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
",
    "path": "dummy-styles/aliases.scss.d.ts",
  },
  Object {
    "contents": "export type Styles = {
  nestedAnother: string;
  nestedClass: string;
  number1: string;
  someStyles: string;
  whereSelector: string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
",
    "path": "dummy-styles/complex.scss.d.ts",
  },
  Object {
    "contents": "export type Styles = {
  composedClass: string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
",
    "path": "dummy-styles/composes.scss.d.ts",
  },
  Object {
    "contents": "export type Styles = {
  app: string;
  appHeader: string;
  logo: string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
",
    "path": "dummy-styles/dashes.scss.d.ts",
  },
  Object {
    "contents": "export type Styles = {
  globalStyle: string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
",
    "path": "dummy-styles/global-variables.scss.d.ts",
  },
  Object {
    "contents": "export type Styles = {
  randomClass: string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
",
    "path": "dummy-styles/invalid.scss.d.ts",
  },
  Object {
    "contents": "export type Styles = {
  nestedStyles: string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
",
    "path": "dummy-styles/nested-styles/style.scss.d.ts",
  },
  Object {
    "contents": "export type Styles = {
  someClass: string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
",
    "path": "dummy-styles/style.scss.d.ts",
  },
]
`;
